test/unit/tech/tech.test.js

```diff
@@ -147,6 +147,97 @@ test('dispose() should clear all tracks that are added after creation', function
   equal(tech.textTracks().length, 0, 'should have zero video tracks after dispose');
 });
 
+test('setSource after tech dispose should dispose source handler once', function(){
+  let MyTech = extendFn(Tech);
+  Tech.withSourceHandlers(MyTech);
+
+  let disposeCount = 0;
+  let handler = {
+    dispose() {
+      disposeCount++;
+    }
+  };
+
+  MyTech.registerSourceHandler({
+    canPlayType: function() {
+      return true;
+    },
+    canHandleSource: function() {
+      return true;
+    },
+    handleSource: function(source, tech, options) {
+      return handler;
+    }
+  });
+
+  let tech = new MyTech();
+  tech.setSource({
```

Comment:
we should set a non-known source here. Here's the test for the previous work: https://github.com/BrandonOCasey/video.js/blob/67e294f0faf8b52e5e88de4576500d1d465e4bee/test/unit/tech/tech.test.js#L453-L495